src/components/Spawn/spawnVolumeModal/getFormSchema.tsx

```diff
@@ -0,0 +1,177 @@
+import { css } from "@emotion/react";
+import { add } from "date-fns";
+import { GetFormSchema } from "components/SpruceForm/types";
+import { ExpirationRow } from "../ExpirationRow";
+import { getDefaultExpiration } from "../utils";
+
+interface Props {
+  maxSpawnableLimit: number;
+  availabilityZones: string[];
+  types: string[];
+  hosts: { id: string; displayName: string }[];
+  disableExpirationCheckbox: boolean;
+  noExpirationCheckboxTooltip: string;
+}
+
+export const getFormSchema = ({
+  maxSpawnableLimit,
+  availabilityZones,
+  types,
+  hosts,
+  disableExpirationCheckbox,
+  noExpirationCheckboxTooltip,
+}: Props): ReturnType<GetFormSchema> => ({
+  fields: {},
+  schema: {
+    type: "object" as "object",
+    properties: {
+      requiredVolumeInformation: {
+        type: "object" as "object",
+        title: "Required Volume Information",
+        properties: {
+          volumeSize: {
+            title: "Size (GB)",
+            type: "number" as "number",
+            default: maxSpawnableLimit > 500 ? 500 : maxSpawnableLimit,
+            minimum: 1,
+            maximum: maxSpawnableLimit,
+          },
+          availabilityZone: {
+            title: "Region",
+            type: "string" as "string",
+            default: availabilityZones?.[0] ?? "",
+            oneOf: availabilityZones.map((r) => ({
+              type: "string" as "string",
+              title: r,
+              enum: [r],
+            })),
+          },
+          type: {
+            title: "Type",
+            type: "string" as "string",
+            default: types?.[0] ?? "",
+            oneOf: types.map((t) => ({
+              type: "string" as "string",
+              title: t,
+              enum: [t],
+            })),
+          },
+        },
+      },
+      optionalVolumeInformation: {
+        type: "object" as "object",
+        title: "Optional Volume Information",
+        properties: {
+          expirationDetails: {
```

Comment:
[nit] It seems like expiration details... aren't really optional? Users need to select either a date or "never expire". Should we bump this above the optional heading?

Reply:
I guess the original intention is that it's optional because it's not really a vital configuration option for the volume. I'd be inclined to leave it, because the spawn host modal also has expiration underneath the optional heading 🤔